---
title: "Correlations"
author: "Timothy Yu"
output: 
  github_document:
    toc: true
    html_preview: false
editor_options:
  chunk_output_type: inline
---
This notebook generates correlations between different datasets.
```{r, include = FALSE}
library(tidyverse)
library(viridis)
library(RColorBrewer)
require(cowplot)
theme_set(theme_cowplot())
library(WGCNA)
```

```{r}
sessionInfo()
```

```{r}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(dev.args = list(png = list(type = "cairo")))
```

Load in datasets.
```{r}
# load in liver lipid data
liver_lipids = read.csv('../processed_data/datasets/Lipidomics_liver.csv', header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)

# load in plasma lipid data
plasma_lipids = read.csv('../processed_data/datasets/Lipidomics_plasma.csv', header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)

# load in RNA-seq FPKM data
RNAseq = read.csv('../processed_data/datasets/RNAseq_FPKM.csv', header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)

# load in intestinal bile acids data
intestinal_ba = read.csv('../processed_data/datasets/BA_intestine.csv', header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)

# load in liver bile acids data
liver_ba = read.csv('../processed_data/datasets/BA_liver.csv', header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)

# load in biliary bile acids data
biliary_ba = read.csv('../processed_data/datasets/BA_biliary.csv', header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)

# load in total bile acids data
total_ba = read.csv('../processed_data/datasets/BA_totalpool.csv', header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)

# load in microbiome (rarefied) data
microbiome = read.csv('../processed_data/datasets/Microbiome_otu_rarefied.csv', header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)
```

```{r}
createCorrTable <- function(corr_object, data1_name, data2_name) {
  "
  Create correlation / BH-adjusted p-value table from bicor object.
  
  Args ---
  corr_object : bicor object that contains bicor and p 
  data1_name : data type of first argument in bicorAndPValue()
  data2_name : data type of second argument in bicorAndPValue()
  "
  bicor <- corr_object$bicor
  p <- corr_object$p

  x <- bicor %>% as.data.frame() %>%
  tibble::rownames_to_column(var = data1_name) %>%
  gather(!!data2_name, bicor, 2:ncol(.))
    
  y <- p %>% as.data.frame() %>%
  tibble::rownames_to_column(var = data1_name) %>%
  gather(!!data2_name, pval, 2:ncol(.)) %>%
  mutate(adj_pval = p.adjust(pval, method = 'BH', n = length(pval)))

  table <- left_join(x, y, by = c(data1_name, data2_name)) %>%
    dplyr::select(!!data1_name, !!data2_name, bicor, adj_pval)
  return(table)
}
```

Correlate liver lipids -> RNA-seq
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(liver_lipids$ID, RNAseq$ID)
liver_lipids_mat = liver_lipids %>% filter(ID %in% common_ids) %>% arrange(ID)
RNAseq_mat = RNAseq %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(liver_lipids_mat$ID, RNAseq_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((RNAseq_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (liver_lipids_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
gene_x_liver_lipid = createCorrTable(corr, 'Gene', 'Liver_lipid') %>%
  filter(adj_pval < 0.05)
write.csv(gene_x_liver_lipid, '../processed_data/correlations/gene_x_liver_lipid_adjpval0.05.csv', row.names = FALSE)

top50 = (gene_x_liver_lipid %>% 
  arrange(adj_pval) %>% 
  dplyr::select(Gene) %>% 
  distinct() %>% 
  head(50))$Gene
```

Correlate liver lipids -> top 50 RNAseq
```{r, warning=FALSE}
top50_RNAseq_mat = RNAseq_mat %>% dplyr::select(ID, Treatment, all_of(top50))

# check that order of ID's are the same in both data sets
assertthat::are_equal(liver_lipids_mat$ID, top50_RNAseq_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((top50_RNAseq_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (liver_lipids_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
top50gene_x_liver_lipid = createCorrTable(corr, 'Gene', 'Liver_lipid')
write.csv(top50gene_x_liver_lipid, '../processed_data/correlations/top50gene_x_liver_lipid.csv', row.names = FALSE)
rm(gene_x_liver_lipid, liver_lipids_mat, RNAseq_mat, top50_RNAseq_mat, top50gene_x_liver_lipid)
```

Correlate liver lipids -> intestinal BA
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(liver_lipids$ID, intestinal_ba$ID)
liver_lipids_mat = liver_lipids %>% filter(ID %in% common_ids) %>% arrange(ID)
intestinal_ba_mat = intestinal_ba %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(liver_lipids_mat$ID, intestinal_ba_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((liver_lipids_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (intestinal_ba_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
liver_lipid_x_intestinal_ba = createCorrTable(corr, 'Liver_lipid', 'Intestinal_ba')
write.csv(liver_lipid_x_intestinal_ba, '../processed_data/correlations/liver_lipid_x_intestinal_ba.csv',
          row.names = FALSE)
rm(liver_lipid_x_intestinal_ba, liver_lipids_mat, intestinal_ba_mat)
```

Correlate liver lipids -> liver BA
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(liver_lipids$ID, liver_ba$ID)
liver_lipids_mat = liver_lipids %>% filter(ID %in% common_ids) %>% arrange(ID)
liver_ba_mat = liver_ba %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(liver_lipids_mat$ID, liver_ba_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((liver_lipids_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (liver_ba_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
liver_lipid_x_liver_ba = createCorrTable(corr, 'Liver_lipid', 'Liver_ba')
write.csv(liver_lipid_x_liver_ba, '../processed_data/correlations/liver_lipid_x_liver_ba.csv',
          row.names = FALSE)
rm(liver_lipid_x_liver_ba, liver_lipids_mat, liver_ba_mat)
```

Correlate liver lipids -> biliary BA
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(liver_lipids$ID, biliary_ba$ID)
liver_lipids_mat = liver_lipids %>% filter(ID %in% common_ids) %>% arrange(ID)
biliary_ba_mat = biliary_ba %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(liver_lipids_mat$ID, biliary_ba_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((liver_lipids_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (biliary_ba_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
liver_lipid_x_biliary_ba = createCorrTable(corr, 'Liver_lipid', 'Biliary_ba')
write.csv(liver_lipid_x_biliary_ba, '../processed_data/correlations/liver_lipid_x_biliary_ba.csv',
          row.names = FALSE)
rm(liver_lipid_x_biliary_ba, liver_lipids_mat, biliary_ba_mat)
```

Correlate liver lipids -> microbiome
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(liver_lipids$ID, microbiome$ID)
liver_lipids_mat = liver_lipids %>% filter(ID %in% common_ids) %>% arrange(ID)
microbiome_mat = microbiome %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(liver_lipids_mat$ID, microbiome_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((liver_lipids_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (microbiome_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
liver_lipid_x_microbiome = createCorrTable(corr, 'Liver_lipid', 'Microbiome_otu')
write.csv(liver_lipid_x_microbiome, '../processed_data/correlations/liver_lipid_x_microbiome.csv',
          row.names = FALSE)
rm(liver_lipid_x_microbiome, liver_lipids_mat, microbiome_mat)
```

Correlate plasma lipids -> RNA-seq
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(plasma_lipids$ID, RNAseq$ID)
plasma_lipids_mat = plasma_lipids %>% filter(ID %in% common_ids) %>% arrange(ID)
RNAseq_mat = RNAseq %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(plasma_lipids_mat$ID, RNAseq_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((RNAseq_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (plasma_lipids_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
gene_x_plasma_lipid = createCorrTable(corr, 'Gene', 'Plasma_lipid') %>%
  filter(adj_pval < 0.05)
write.csv(gene_x_plasma_lipid, '../processed_data/correlations/gene_x_plasma_lipid_adjpval0.05.csv', 
          row.names = FALSE)
rm(gene_x_plasma_lipid, plasma_lipids_mat, RNAseq_mat)
```

Correlate plasma lipids -> intestinal BA
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(plasma_lipids$ID, intestinal_ba$ID)
plasma_lipids_mat = plasma_lipids %>% filter(ID %in% common_ids) %>% arrange(ID)
intestinal_ba_mat = intestinal_ba %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(plasma_lipids_mat$ID, intestinal_ba_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((plasma_lipids_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (intestinal_ba_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
plasma_lipid_x_intestinal_ba = createCorrTable(corr, 'Plasma_lipid', 'Intestinal_ba')
write.csv(plasma_lipid_x_intestinal_ba,'../processed_data/correlations/plasma_lipid_x_intestinal_ba.csv',
          row.names = FALSE)
rm(plasma_lipid_x_intestinal_ba, plasma_lipids_mat, intestinal_ba_mat)
```

Correlate plasma lipids -> liver BA
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(plasma_lipids$ID, liver_ba$ID)
plasma_lipids_mat = plasma_lipids %>% filter(ID %in% common_ids) %>% arrange(ID)
liver_ba_mat = liver_ba %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(plasma_lipids_mat$ID, liver_ba_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((plasma_lipids_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (liver_ba_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
plasma_lipid_x_liver_ba = createCorrTable(corr, 'Plasma_lipid', 'Liver_ba')
write.csv(plasma_lipid_x_liver_ba,'../processed_data/correlations/plasma_lipid_x_liver_ba.csv',
          row.names = FALSE)
rm(plasma_lipid_x_liver_ba, plasma_lipids_mat, liver_ba_mat)
```

Correlate plasma lipids -> biliary BA
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(plasma_lipids$ID, biliary_ba$ID)
plasma_lipids_mat = plasma_lipids %>% filter(ID %in% common_ids) %>% arrange(ID)
biliary_ba_mat = biliary_ba %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(plasma_lipids_mat$ID, biliary_ba_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((plasma_lipids_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (biliary_ba_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
plasma_lipid_x_biliary_ba = createCorrTable(corr, 'Plasma_lipid', 'Biliary_ba')
write.csv(plasma_lipid_x_biliary_ba,'../processed_data/correlations/plasma_lipid_x_biliary_ba.csv',
          row.names = FALSE)
rm(plasma_lipid_x_biliary_ba, plasma_lipids_mat, biliary_ba_mat)
```

Correlate plasma lipids -> microbiome
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(plasma_lipids$ID, microbiome$ID)
plasma_lipids_mat = plasma_lipids %>% filter(ID %in% common_ids) %>% arrange(ID)
microbiome_mat = microbiome %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(plasma_lipids_mat$ID, microbiome_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((plasma_lipids_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (microbiome_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
plasma_lipid_x_microbiome = createCorrTable(corr, 'Plasma_lipid', 'Microbiome_otu')
write.csv(plasma_lipid_x_microbiome,'../processed_data/correlations/plasma_lipid_x_microbiome.csv',
          row.names = FALSE)
rm(plasma_lipid_x_microbiome, plasma_lipids_mat, microbiome_mat)
```

Correlate microbiome -> RNA-seq
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(microbiome$ID, RNAseq$ID)
microbiome_mat = microbiome %>% filter(ID %in% common_ids) %>% arrange(ID)
RNAseq_mat = RNAseq %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(microbiome_mat$ID, RNAseq_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((RNAseq_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (microbiome_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
gene_x_microbiome = createCorrTable(corr, 'Gene', 'Microbiome_otu') %>%
  filter(adj_pval < 0.05)
write.csv(gene_x_microbiome, '../processed_data/correlations/gene_x_microbiome_adjpval0.05.csv', row.names = FALSE)
rm(gene_x_microbiome, microbiome_mat, RNAseq_mat)
```

Correlate microbiome -> intestinal BA
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(microbiome$ID, intestinal_ba$ID)
microbiome_mat = microbiome %>% filter(ID %in% common_ids) %>% arrange(ID)
intestinal_ba_mat = intestinal_ba %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(microbiome_mat$ID, intestinal_ba_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((intestinal_ba_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (microbiome_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
intestinal_ba_x_microbiome = createCorrTable(corr, 'Intestinal_ba', 'Microbiome_otu')
write.csv(intestinal_ba_x_microbiome, '../processed_data/correlations/intestinal_ba_x_microbiome.csv',
          row.names = FALSE)
rm(intestinal_ba_x_microbiome, microbiome_mat, intestinal_ba_mat)
```

Correlate microbiome -> liver BA
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(microbiome$ID, liver_ba$ID)
microbiome_mat = microbiome %>% filter(ID %in% common_ids) %>% arrange(ID)
liver_ba_mat = liver_ba %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(microbiome_mat$ID, liver_ba_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((liver_ba_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (microbiome_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
liver_ba_x_microbiome = createCorrTable(corr, 'Liver_ba', 'Microbiome_otu')
write.csv(liver_ba_x_microbiome, '../processed_data/correlations/liver_ba_x_microbiome.csv',
          row.names = FALSE)
rm(liver_ba_x_microbiome, microbiome_mat, liver_ba_mat)
```

Correlate microbiome -> biliary BA
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(microbiome$ID, biliary_ba$ID)
microbiome_mat = microbiome %>% filter(ID %in% common_ids) %>% arrange(ID)
biliary_ba_mat = biliary_ba %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(microbiome_mat$ID, biliary_ba_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((biliary_ba_mat %>% dplyr::select(-c('ID', 'Treatment'))),
                      (microbiome_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
biliary_ba_x_microbiome = createCorrTable(corr, 'Biliary_ba', 'Microbiome_otu')
write.csv(biliary_ba_x_microbiome, '../processed_data/correlations/biliary_ba_x_microbiome.csv',
          row.names = FALSE)
rm(biliary_ba_x_microbiome, microbiome_mat, biliary_ba_mat)
```

Correlate microbiome -> total BA
```{r, warning=FALSE}
# only correlate if ID's are in both data sets
common_ids = intersect(microbiome$ID, total_ba$ID)
microbiome_mat = microbiome %>% filter(ID %in% common_ids) %>% arrange(ID)
total_ba_mat = total_ba %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(microbiome_mat$ID, total_ba_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((total_ba_mat %>% dplyr::select(-c('ID', 'Treatment'))),
                      (microbiome_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )
```

```{r}
total_ba_x_microbiome = createCorrTable(corr, 'Total_ba', 'Microbiome_otu')
write.csv(total_ba_x_microbiome, '../processed_data/correlations/total_ba_x_microbiome.csv',
          row.names = FALSE)
rm(total_ba_x_microbiome, microbiome_mat, total_ba_mat)
```

```{r}
# rmarkdown::render("correlations.Rmd")
# mv correlations.md ../markdowns/
```





















