---
title: "Microbiome analysis"
author: "Timothy Yu"
date: "3/27/2021"
output: html_document
---
```{r, include = FALSE}
library(tidyverse)
library(ggplot2)
library(ggfortify)
require(cowplot)
library(WGCNA)
theme_set(theme_cowplot())
```

```{r}
# using the rarefied data for these analyses
data = read.csv('../processed_data/datasets/Microbiome_otu_rarefied.csv', header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)
```

```{r}
# PCA
autoplot(prcomp(data[-c(1,2)], scale. = TRUE), data = data, colour = 'Treatment', size = 2.5) + scale_color_manual(values = c('black', '#3b9bb3', '#75af4f', '#e39225', '#c13b41'))
ggsave('../figures/microbiome_rarefied_pca.pdf', height = 3.5, width = 5)
```
```{r}
# load in taxonomic information
tax_info = read.csv('../raw_data/Mappings/Microbiome_tax_class_rarefied.csv', header = TRUE,
                    stringsAsFactors = FALSE)
```

  Rank1     Rank2    Rank3     Rank4     Rank5     Rank6     Rank7
"Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"
```{r}
total_otus = rowSums(data[-c(1,2)])
# because rarefied, all have 73989 reads

data %>%
  dplyr::select(ID, Treatment, (tax_info %>% filter(Rank5 == 'f__Verrucomicrobiaceae'))$OTU) %>%
  mutate(percentVerruco = ((dplyr::select(., -c(ID, Treatment)) %>% rowSums) / 73989) * 100) %>%
  dplyr::select(ID, Treatment, percentVerruco) %>%
  ggplot(aes(x = Treatment, y = percentVerruco)) + geom_boxplot(outlier.shape=NA) +
  geom_jitter(aes(color = Treatment), size = 3) +
  scale_color_manual(values = c('black', '#3b9bb3', '#75af4f', '#e39225', '#c13b41')) +
  labs(x = 'Diet', y = 'Relative Abundance Per Sample (%)', title = 'f_Verrucomicrobioaceae')
```
```{r}
# load in total BA
BA_totalpool = read.csv('../processed_data/datasets/BA_totalpool.csv', header = TRUE,
                        stringsAsFactors = FALSE, check.names = FALSE)
```

```{r}
createCorrTable <- function(corr_object, data1_name, data2_name) {
  "
  Create correlation / BH-adjusted p-value table from bicor object.
  
  Args ---
  corr_object : bicor object that contains bicor and p 
  data1_name : data type of first argument in bicorAndPValue()
  data2_name : data type of second argument in bicorAndPValue()
  "
  bicor <- corr_object$bicor
  p <- corr_object$p

  x <- bicor %>% as.data.frame() %>%
  tibble::rownames_to_column(var = data1_name) %>%
  gather(!!data2_name, bicor, 2:ncol(.))
    
  y <- p %>% as.data.frame() %>%
  tibble::rownames_to_column(var = data1_name) %>%
  gather(!!data2_name, pval, 2:ncol(.)) %>%
  mutate(adj_pval = p.adjust(pval, method = 'BH', n = length(pval)))

  table <- left_join(x, y, by = c(data1_name, data2_name)) %>%
    dplyr::select(!!data1_name, !!data2_name, bicor, adj_pval)
  return(table)
}
```

```{r}
# correlation between firmicutes and total BA species
firmicutes = data %>% 
  dplyr::select(ID, Treatment, (tax_info %>% filter(Rank2 == 'p__Firmicutes'))$OTU)

# only correlate if ID's are in both data sets
common_ids = intersect(firmicutes$ID, BA_totalpool$ID)
firmicutes_mat = firmicutes %>% filter(ID %in% common_ids) %>% arrange(ID)
BA_totalpool_mat = BA_totalpool %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(firmicutes_mat$ID, BA_totalpool_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((BA_totalpool_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (firmicutes_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )

firmicutes_x_BA_totalpool = createCorrTable(corr, 'totalpool_ba', 'firmicutes')
```
```{r}
# plot heatmap of significant correlations
sig_firmicutes = (firmicutes_x_BA_totalpool %>% filter(adj_pval < 0.05))$firmicutes %>% unique()

# selected data
selected_df = firmicutes_x_BA_totalpool %>%
  filter(firmicutes %in% sig_firmicutes) %>%
  dplyr::select(-adj_pval) %>%
  spread(firmicutes, bicor, fill=0) %>%
  tibble::column_to_rownames(var="totalpool_ba")

# hierarchical clustering for order
ba_ord = hclust(dist(selected_df, method = "euclidean"), method = "ward.D")$order
firmicutes_ord = hclust(dist(t(selected_df), method = "euclidean"), method = "ward.D")$order

# plot
firmicutes_x_BA_totalpool %>%
  filter(firmicutes %in% sig_firmicutes) %>%
  ggplot(aes(x = firmicutes, y = totalpool_ba)) + geom_tile(aes(fill = bicor)) +
  scale_fill_distiller(palette='RdYlBu') + 
  scale_y_discrete(limits=rownames(selected_df)[ba_ord]) + 
  scale_x_discrete(limits=colnames(selected_df)[firmicutes_ord]) +
  theme(axis.ticks.x = element_blank(),
        axis.text.x = element_blank()) +
  labs(x="Firmicutes species", y="Bile Acids (Total Pool)")
ggsave('../figures/microbiome_firmicutes_x_ba_totalpool.pdf', height = 4, width = 6)
```

```{r}
# correlation between Bacteroidetes and total BA species
bacteroidetes = data %>% 
  dplyr::select(ID, Treatment, (tax_info %>% filter(Rank2 == 'p__Bacteroidetes'))$OTU)

# only correlate if ID's are in both data sets
common_ids = intersect(bacteroidetes$ID, BA_totalpool$ID)
bacteroidetes_mat = bacteroidetes %>% filter(ID %in% common_ids) %>% arrange(ID)
BA_totalpool_mat = BA_totalpool %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(bacteroidetes_mat$ID, BA_totalpool_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((BA_totalpool_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (bacteroidetes_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )

bacteroidetes_x_BA_totalpool = createCorrTable(corr, 'totalpool_ba', 'bacteroidetes')
```
```{r}
# plot heatmap of significant correlations
sig_bacteroidetes = (bacteroidetes_x_BA_totalpool %>% filter(adj_pval < 0.05))$bacteroidetes %>% unique()

# selected data
selected_df = bacteroidetes_x_BA_totalpool %>%
  filter(bacteroidetes %in% sig_bacteroidetes) %>%
  dplyr::select(-adj_pval) %>%
  spread(bacteroidetes, bicor, fill=0) %>%
  tibble::column_to_rownames(var="totalpool_ba")

# hierarchical clustering for order
ba_ord = hclust(dist(selected_df, method = "euclidean"), method = "ward.D")$order
bacteroidetes_ord = hclust(dist(t(selected_df), method = "euclidean"), method = "ward.D")$order

# plot
bacteroidetes_x_BA_totalpool %>%
  filter(bacteroidetes %in% sig_bacteroidetes) %>%
  ggplot(aes(x = bacteroidetes, y = totalpool_ba)) + geom_tile(aes(fill = bicor)) +
  scale_fill_distiller(palette='RdYlBu') + 
  scale_y_discrete(limits=rownames(selected_df)[ba_ord]) + 
  scale_x_discrete(limits=colnames(selected_df)[bacteroidetes_ord]) +
  theme(axis.ticks.x = element_blank(),
        axis.text.x = element_blank()) +
  labs(x="Bacteroidetes species", y="Bile Acids (Total Pool)")
ggsave('../figures/microbiome_bacteroidetes_x_ba_totalpool.pdf', height = 4, width = 6)
```
```{r}
# correlation between Verrucomicrobia and total BA species
verruco = data %>% 
  dplyr::select(ID, Treatment, (tax_info %>% filter(Rank2 == 'p__Verrucomicrobia'))$OTU)

# only correlate if ID's are in both data sets
common_ids = intersect(verruco$ID, BA_totalpool$ID)
verruco_mat = verruco %>% filter(ID %in% common_ids) %>% arrange(ID)
BA_totalpool_mat = BA_totalpool %>% filter(ID %in% common_ids) %>% arrange(ID)

# check that order of ID's are the same in both data sets
assertthat::are_equal(verruco_mat$ID, BA_totalpool_mat$ID)

# correlate (NA's and missing values exist, so warnings expected)
corr = bicorAndPvalue((BA_totalpool_mat %>% dplyr::select(-c('ID', 'Treatment'))), 
                      (verruco_mat %>% dplyr::select(-c('ID', 'Treatment')))
       )

verruco_x_BA_totalpool = createCorrTable(corr, 'totalpool_ba', 'verruco')
```

```{r}
# plot heatmap of significant correlations
sig_verruco = (verruco_x_BA_totalpool %>% filter(adj_pval < 0.05))$verruco %>% unique()

# selected data
selected_df = verruco_x_BA_totalpool %>%
  filter(verruco %in% sig_verruco) %>%
  dplyr::select(-adj_pval) %>%
  spread(verruco, bicor, fill=0) %>%
  tibble::column_to_rownames(var="totalpool_ba")

# hierarchical clustering for order
ba_ord = hclust(dist(selected_df, method = "euclidean"), method = "ward.D")$order
verruco_ord = hclust(dist(t(selected_df), method = "euclidean"), method = "ward.D")$order

# plot
verruco_x_BA_totalpool %>%
  filter(verruco %in% sig_verruco) %>%
  ggplot(aes(x = verruco, y = totalpool_ba)) + geom_tile(aes(fill = bicor)) +
  scale_fill_distiller(palette='RdYlBu') + 
  scale_y_discrete(limits=rownames(selected_df)[ba_ord]) + 
  scale_x_discrete(limits=colnames(selected_df)[verruco_ord]) +
  theme(axis.ticks.x = element_blank(),
        axis.text.x = element_blank()) +
  labs(x="Verrucomicrobia species", y="Bile Acids (Total Pool)")
ggsave('../figures/microbiome_verruco_x_ba_totalpool.pdf', height = 4, width = 6)
```
```{r}
# positive correlation plot between single bacteriodetes species and T-b-MCA
r = cor(
  bacteroidetes_mat$`4457872`, 
  BA_totalpool_mat$`T-b-MCA`,  
  method = "pearson"
)

ggplot(data=NULL,
  aes(x=bacteroidetes_mat$`4457872`, y=BA_totalpool_mat$`T-b-MCA`)) +
  geom_smooth(method='lm', se=FALSE) +
  geom_point(size=2) + 
  annotate("text", x=5000, y=20, size=5, label="italic(r) == 0.76", parse=TRUE) +
  labs(y='T-b-MCA (nmol)', x='bacteriodetes species 4457872 (rarefied counts)')
ggsave('../figures/microbiome_bacteroidete_species_x_TbMCA_pos.pdf', height = 4.3, width = 5)
```

```{r}
# negative correlation plot between single bacteriodetes species and T-b-MCA
r = cor(
  bacteroidetes_mat$`183446`, 
  BA_totalpool_mat$`T-b-MCA`,  
  method = "pearson"
)

ggplot(data=NULL,
  aes(x=bacteroidetes_mat$`183446`, y=BA_totalpool_mat$`T-b-MCA`)) +
  geom_smooth(method='lm', se=FALSE) +
  geom_point(size=2) + 
  annotate("text", x=4000, y=20, size=5, label="italic(r) == -0.64", parse=TRUE) +
  labs(y='T-b-MCA (nmol)', x='bacteriodetes species 183446 (rarefied counts)')
ggsave('../figures/microbiome_bacteroidete_species_x_TbMCA_neg.pdf', height = 4.3, width = 5)
```

```{r}
# positive correlation plot between single verrucomicrobia species and T-b-MCA
r = cor(
  verruco_mat$`4306262`, 
  BA_totalpool_mat$`T-b-MCA`,  
  method = "pearson"
)

ggplot(data=NULL,
  aes(x=verruco_mat$`4306262`, y=BA_totalpool_mat$`T-b-MCA`)) +
  geom_smooth(method='lm', se=FALSE) +
  geom_point(size=2) + 
  annotate("text", x=5000, y=20, size=5, label="italic(r) == 0.59", parse=TRUE) +
  labs(y='T-b-MCA (nmol)', x='verrucomicrobia species 4306262 (rarefied counts)')
ggsave('../figures/microbiome_verruco_species_x_TbMCA_pos_1.pdf', height = 4.3, width = 5)
```
```{r}
# positive correlation plot between single verrucomicrobia species and T-b-MCA
r = cor(
  verruco_mat$`359376`, 
  BA_totalpool_mat$`T-b-MCA`,  
  method = "pearson"
)

ggplot(data=NULL,
  aes(x=verruco_mat$`359376`, y=BA_totalpool_mat$`T-b-MCA`)) +
  geom_smooth(method='lm', se=FALSE) +
  geom_point(size=2) + 
  annotate("text", x=3, y=20, size=5, label="italic(r) == 0.46", parse=TRUE) +
  labs(y='T-b-MCA (nmol)', x='verrucomicrobia species 4306262 (rarefied counts)')
ggsave('../figures/microbiome_verruco_species_x_TbMCA_pos_2.pdf', height = 4.3, width = 5)
```