```{r, include = FALSE}
library(dplyr)
library(tidyr)
library(ggplot2)
require(cowplot)
theme_set(theme_cowplot())
library(viridis)
library(RColorBrewer)
library(DESeq2)
library(GenomicFeatures)
library(apeglm)
library(EnhancedVolcano)
```

Load in all sample files.
Note: one outlier `7B_Index_09_control8` was removed after visualizing PCA. If want to visualize it without that sample, do not run line of code where it removes file name from `sampleFiles`.
```{r}
directory <- "../raw_data/ETV007_RNAseq_counts"
sampleFiles <- (list.files(directory))
sampleFiles <- sampleFiles[-33]

# load in the gene key information
key = read.table('../raw_data/Mappings/ensembl84_info.txt', header = TRUE, sep = '\t', quote="", fill=FALSE, stringsAsFactors = FALSE)
```

Get gene lengths for downstream FPKM normalization.
```{r}
gene_info = key %>% mutate(gene_length = Gene.End..bp. - Gene.Start..bp. + 1) %>% dplyr::select(Ensembl.Gene.ID, Associated.Gene.Name, gene_length) %>% distinct()
```

Create `DESeq` object.
```{r}
sampleCondition <- sub("(.*treated).*","\\1",sampleFiles)
sampleTable <- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleFiles)

sampleTable <- sampleTable %>% 
  separate(col = 'condition', into = c("exp", "index", "condition"), sep = "_", remove = F)
sampleTable <- subset(sampleTable, select = -c(exp, index))
sampleTable$condition <- gsub(".count", "", sampleTable$condition)
sampleTable$condition <- gsub("control8", "0", sampleTable$condition)
sampleTable$condition <- gsub("treated", "", sampleTable$condition)

ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                       directory = directory,
                                       design = ~condition)

#Set the reference group to 0 week
ddsHTSeq$condition <- relevel(ddsHTSeq$condition, ref = "0")

#Filter out reads whose row sums are less than 10
ddsHTSeq <- ddsHTSeq[rowSums(counts(ddsHTSeq))>=10,]

# Run DESeq
dds <- DESeq(ddsHTSeq)
```

PCA of variance-stabilized transformed data.
```{r}
vsd = vst(dds)
levels(vsd$condition) = c("CHOW8", "WD2", "WD4", "WD6", "WD8")
plotPCA(vsd) + 
  scale_color_manual(values = c('black', '#3b9bb3', '#75af4f', '#e39225', '#c13b41'), 
                     name = "Treatment")
ggsave('../figures/rnaseq_pca.pdf', height = 4.5, width = 5)
```
Check normalized data.
```{r}
raw <- reshape2::melt(assay(dds))
colnames(raw) <- c("Gene","Sample","Expression")
ggplot(raw, aes(x=Sample, y=Expression)) + geom_violin() + theme(axis.text.x = element_text(angle=45, hjust=1))

matrix <- reshape2::melt(assay(vsd))
colnames(matrix) <- c("Gene","Sample","Expression")
ggplot(matrix, aes(x=Sample, y=Expression)) + geom_violin() + theme(axis.text.x = element_text(angle=45, hjust=1))
```

Generate all volcano plots. We use `ashr` to shrink fold-changes.
Vertical lines indicate |log2FC| > 0.5.
Horizontal lines indicate padj < 0.05.
```{r}
mapping = key %>% dplyr::select(Ensembl.Gene.ID, Associated.Gene.Name) %>% distinct()

for(i in c(0, 2, 4, 6)) {
  for (j in c(2, 4, 6, 8)) {
    if(j > i) {
      res = lfcShrink(dds, contrast = c("condition", j, i), type = 'ashr')
      annotation = data.frame("Ensembl.Gene.ID" = rownames(res), stringsAsFactors = FALSE) %>% 
      left_join(., mapping, by = "Ensembl.Gene.ID") 
      
      # create custom key-value pairs for 'up-regulated', 'down-regulated', 'non-significant'
      keyvals <- ifelse(
        res$log2FoldChange > 0.5 & res$padj < 0.05, 'blue', ifelse(
          res$log2FoldChange < -0.5 & res$padj < 0.05, 'red', 'gray40'
        )
      )
      keyvals[is.na(keyvals)] <- 'gray40'
      names(keyvals)[keyvals == 'blue'] <- 'up-regulated'
      names(keyvals)[keyvals == 'red'] <- 'down-regulated'
      names(keyvals)[keyvals == 'gray40'] <- 'NS'
      
      EnhancedVolcano(res, 
                lab = annotation$Associated.Gene.Name, 
                labSize = 6,
                x = 'log2FoldChange', 
                y = 'padj', 
                title = paste(i, 'vs.', j), 
                pCutoff = 0.05, 
                FCcutoff = 0.5, 
                colCustom = keyvals,
                colAlpha = 0.9, 
                pointSize = 3.0)
      ggsave(paste('../figures/volcano/', i, 'vs', j, '.pdf', sep=''), height = 7.5, width = 6.2)
    }
  }
}
```

Obtain log2FC, padj, and direction of FC for each gene between each time-point comparison.
Calculate the number of DE genes between each time-point comparison.
DE means padj < 0.05 and |log2FC| > 0.5.
```{r}
mapping = key %>% dplyr::select(Ensembl.Gene.ID, Associated.Gene.Name) %>% distinct()
class_df = data.frame(matrix(NA, ncol=1, nrow=nrow(annotation)))[-1]
summary_df = data.frame("t1"=integer(), "t2"=integer(), "Total" =integer(), 
                        "Up"=integer(), "Down"=integer(), stringsAsFactors=FALSE)

for(i in c(0, 2, 4, 6)) {
  for (j in c(2, 4, 6, 8)) {
    if(j > i) {
      res = lfcShrink(dds, contrast = c("condition", j, i), type = 'ashr')
      fcname = paste('log2FC_', i, j, sep='')
      dename = paste('DE_', i, j, sep='')
      dirname = paste('Direct_', i, j, sep='')
  
      genes = as.data.frame(res) %>% 
        tibble::rownames_to_column(var = "Ensembl.Gene.ID") %>% 
        left_join(., mapping, by = "Ensembl.Gene.ID") %>%
        mutate(
          !!fcname := log2FoldChange,
          !!dename := ifelse(padj < 0.05 & abs(log2FoldChange) > 0.5, "yes", "no"),
          !!dirname := ifelse(log2FoldChange > 0, "up", "down")
        ) %>%
        dplyr::select(Associated.Gene.Name, !!as.symbol(fcname), 
                      !!as.symbol(dename), !!as.symbol(dirname))
  
      if(ncol(class_df) == 0) {
        class_df = cbind(class_df, genes)
      } else {
        stopifnot(identical(class_df$Associated.Gene.Name, genes$Associated.Gene.Name))
        class_df = cbind(class_df, genes %>% dplyr::select(-Associated.Gene.Name))
      }
      
      num_total = nrow(filter(genes, !!as.symbol(dename) == "yes"))
      num_up = nrow(filter(genes, !!as.symbol(dename) == "yes", !!as.symbol(dirname) == "up"))
      num_down = nrow(filter(genes, !!as.symbol(dename) == "yes", !!as.symbol(dirname) == "down"))
      num_category = data.frame("t1"=i, "t2"=j, "Total"=num_total, "Up"=num_up, 
                              "Down"=num_down, stringsAsFactors=FALSE)
      summary_df = rbind(summary_df, num_category)
    }
  }
}
```
Plot heat maps of significantly DE genes (padj < 0.05, |log2FC| > 0.5).
```{r}
a <- summary_df %>% 
  ggplot(aes(x=t1, y=t2)) + 
  geom_tile(aes(fill=Total), color='black', size=0.5, alpha = 0.9) + 
  scale_fill_gradient(low="#f1f1f1", high="#696969") + 
  scale_x_continuous(breaks=seq(0,8,2)) + 
  scale_y_continuous(breaks=seq(0,8,2)) + 
  labs(x="Weeks on Western Diet", y="Weeks on Western Diet", title="Total") + 
  geom_text(aes(label=Total), size=6) + 
  theme(legend.position="none")

b <- summary_df %>% 
  ggplot(aes(x=t1, y=t2)) + 
  geom_tile(aes(fill=Up), color='black', size=0.5, alpha=0.9) +
  scale_fill_gradient(low="#e5e5ff", high="blue") + 
  scale_x_continuous(breaks=seq(0,8,2)) + 
  scale_y_continuous(breaks=seq(0,8,2)) + 
  labs(x="Weeks on Western Diet", y="Weeks on Western Diet", title="Up-regulated") +
  geom_text(aes(label=Up), size=6) + 
  theme(legend.position="none")

c <- summary_df %>% 
  ggplot(aes(x=t1, y=t2)) + 
  geom_tile(aes(fill=Down), color='black', size=0.5, alpha=0.9) +
  scale_fill_gradient(low="#ffe5e5", high="red") +
  scale_x_continuous(breaks=seq(0,8,2)) + 
  scale_y_continuous(breaks=seq(0,8,2)) + 
  labs(x="Weeks on Western Diet", y="Weeks on Western Diet", title="Down-regulated") +
  geom_text(aes(label=Down), size=6) + 
  theme(legend.position="none")

plot_grid(a,b,c, ncol = 3)
ggsave('../figures/rnaseq_pairwise_de_genes_heatmap_fccutoff.pdf', height = 3.7, width = 10)
rm(a,b,c)
```

The RNA-seq profiles have a largely `biphasic` response. We can categorize genes accordingly and count the number in each category.
```{r}
class_df = class_df %>% 
  mutate(biphasic_class = ifelse(DE_02 == 'yes' & DE_28 == 'no', 'uniq_02',
                 ifelse(DE_02 == 'no' & DE_28 == 'yes', 'uniq_28',
                 ifelse(DE_02 == 'yes' & DE_28 == 'yes', 'both', 'none'))))

print(nrow(filter(class_df, biphasic_class == 'uniq_02')))
print(nrow(filter(class_df, biphasic_class == 'uniq_28')))
print(nrow(filter(class_df, biphasic_class == 'both'))) 
print(nrow(filter(class_df, biphasic_class == 'none'))) 

#[1] 659
#[1] 494
#[1] 85
#[1] 21494
```
Save lists of DE genes that fall into each category for downstream KEGG pathway enrichment analysis.
```{r}
save_list <- function(l, fname) {
  write.table(
    l,
    paste0('../processed_data/degenes/', fname, '.txt'),
    quote = F, col.names = F, row.names = F
  )
}

# DE exclusively between 02, up
up_02_list = (
  class_df %>% 
    filter(biphasic_class == 'uniq_02', Direct_02 == 'up') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(up_02_list, 'up_02')

# DE exclusively between 02, down
down_02_list = (
  class_df %>% 
    filter(biphasic_class == 'uniq_02', Direct_02 == 'down') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(down_02_list, 'down_02')

# DE exclusively between 28, up
up_28_list = (
  class_df %>% 
    filter(biphasic_class == 'uniq_28', Direct_28 == 'up') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(up_28_list, 'up_28')

# DE exclusively between 28, down
down_28_list = (
  class_df %>% 
    filter(biphasic_class == 'uniq_28', Direct_28 == 'down') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(down_28_list, 'down_28')

# DE between 02 (up) and 28 (up) 
up02_up28_list = (
  class_df %>% 
    filter(biphasic_class == 'both', Direct_02 == 'up', Direct_28 == 'up') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(up02_up28_list, 'up02_up28')

# DE between 02 (up) and 28 (down) 
up02_down28_list = (
  class_df %>% 
    filter(biphasic_class == 'both', Direct_02 == 'up', Direct_28 == 'down') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(up02_down28_list, 'up02_down28')

# DE between 02 (down) and 28 (up) 
down02_up28_list = (
  class_df %>% 
    filter(biphasic_class == 'both', Direct_02 == 'down', Direct_28 == 'up') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(down02_up28_list, 'down02_up28')

# DE between 02 (down) and 28 (down) 
down02_down28_list = (
  class_df %>% 
    filter(biphasic_class == 'both', Direct_02 == 'down', Direct_28 == 'down') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(down02_down28_list, 'down02_down28')
```

We input the above gene lists into KEGG pathway enrichment via `http://bioinformatics.sdstate.edu/go/` using their default background option: all protein coding genes in the genome.

We then filter out the outputted KEGG pathways by their:
1. FDR < 0.05
2. any obvious irrelevant pathways (i.e., cancer, viral infection, etc.)
3. any super broad categories (i.e., metabolic pathways)
```{r}
kegg_02_up = read.csv('../processed_data/degenes/up_02_kegg.csv', header = TRUE) %>%
  mutate(comparison = 'up_02') %>%
  filter(Enrichment.FDR < 0.05) %>%
  filter(!Pathway %in% c('Coronavirus disease', 'Epstein-Barr virus infection', 
                         'Neutrophil extracellular trap formation', 'Measles',
                         'Osteoclast differentiation', 'Staphylococcus aureus infection', 
                         'Hepatitis C', 'Influenza A',
                         'Proteoglycans in cancer', 'Leishmaniasis', 
                         'Acute myeloid leukemia', 'Cell adhesion molecules',
                         'Hematopoietic cell lineage'))

kegg_02_down = read.csv('../processed_data/degenes/down_02_kegg.csv', header = TRUE) %>%
  mutate(comparison = 'down_02') %>%
  filter(Enrichment.FDR < 0.05) %>%
  filter(!Pathway %in% c('Metabolic pathways', 'Chemical carcinogenesis',
                         'Drug metabolism'))

kegg_28_up = read.csv('../processed_data/degenes/up_28_kegg.csv', header = TRUE) %>%
  mutate(comparison = 'up_28') %>%
  filter(Enrichment.FDR < 0.05) %>%
  filter(!Pathway %in% c('Focal adhesion', 'Axon guidance', 'Human papillomavirus infection',
                         'Amoebiasis', 'MicroRNAs in cancer', 'Malaria', 'Rheumatoid arthritis',
                         'Hepatitis B', 'Proteoglycans in cancer'))

kegg_28_down = read.csv('../processed_data/degenes/down_28_kegg.csv', header = TRUE) %>%
  mutate(comparison = 'down_28') %>%
  filter(Enrichment.FDR < 0.05)

kegg_02down_28up = read.csv('../processed_data/degenes/down02_up28_kegg.csv', header = TRUE) %>%
  mutate(comparison = 'down02_up28') %>%
  filter(Enrichment.FDR < 0.05) %>%
  filter(!Pathway %in% c('Chemical carcinogenesis', 'Drug metabolism', 'Platinum drug resistance',
                         'Metabolic pathways', 'Pathways in cancer'))

kegg_df = rbind(kegg_02_up, kegg_02_down, kegg_28_up, kegg_28_down, kegg_02down_28up)
```

Plot KEGG pathway enrichment.
```{r}
plot_kegg <- function(data, fname, height, width, color) {
  p = ggplot(data=data,
         aes(x = reorder(Pathway, Fold.Enrichment), 
             y = Fold.Enrichment, 
             size = -log10(Enrichment.FDR))) + 
  geom_point(alpha=1, shape=21, color="black", fill = color) +
  scale_size(range = c(1, 5), name="-log10(Enrichment FDR)") +
  coord_flip() +
  labs(y = 'Fold Enrichment') +
  theme(axis.title.y=element_blank()) +
  theme(axis.title.y=element_blank(),
        panel.grid.major.y = element_line(colour = "grey80", linetype = 'dashed')
  ) 
  
  ggsave(plot=p, paste0('../figures/', fname, '.pdf'), height=height, width=width)
}

plot_kegg(kegg_02_up, 'rnaseq_kegg_02_up', height=4, width=8, color='blue')
plot_kegg(kegg_02_down, 'rnaseq_kegg_02_down', height=4, width=8, color='red')
plot_kegg(kegg_28_up, 'rnaseq_kegg_28_up', height=5, width=8, color='blue')
plot_kegg(kegg_28_down, 'rnaseq_kegg_28_down', height=2, width=8, color='red')
plot_kegg(kegg_02down_28up, 'rnaseq_kegg_02down_28up', height=3, width=8, color='#FFA500')
```

```{r}
# how many genes are DE between 0-2, not any comparison after 2 weeks? 643
only_02 = class_df %>% filter(DE_02 == "yes", 
                              DE_24 == "no", DE_26 == "no", DE_28 == "no",
                              DE_46 == "no", DE_48 == "no", DE_68 == "no")
# up? 472
up_only_02 = only_02 %>% filter(Direct_02 == "up")
  
# down? 171
down_only_02 = only_02 %>% filter(Direct_02 == "down")

# how many genes are DE between 2-4, not any comparison before or after that? 4 (2 up, 2 down)
only_24 = class_df %>% filter(DE_24 == "yes", 
                              DE_02 == "no", DE_46 == "no", 
                              DE_48 == "no", DE_68 == "no")

# how many genes are DE between 4-6, not any comparison before or after that? 2 down
only_46 = class_df %>% filter(DE_46 == "yes", 
                              DE_02 == "no", DE_04 == "no", 
                              DE_24 == "no", DE_68 == "no")

# how many genes are DE between 2-8, not any comparison before and must be positively increasing for rest?
only_28 = class_df %>% filter(DE_28 == "yes", 
                              DE_02 == "no",
                              Direct_24 == "up", Direct_46 == "up", Direct_68 == "up")

# how many genes are monotonically increasing in expression? (i.e., up-reguulated DE between 0-2, 2-4, 4-6, 6-8)
monotonic_up = class_df %>% filter(DE_02 == "yes", Direct_02 == "up",
                                   DE_28 == "yes", Direct_28 == "up"
                                   )
```

```{r}
at_least_one_sig_df = class_df %>% filter(
   DE_02 == "yes" | DE_04 == "yes" | DE_06 == "yes" | DE_08 == "yes" |
   DE_24 == "yes" | DE_26 == "yes" | DE_28 == "yes" |
   DE_46 == "yes" | DE_48 == "yes" |
   DE_68 == "yes"
) %>% dplyr::select(Associated.Gene.Name, 
                    DE_02, log2FC_02, Direct_02,
                    DE_24, log2FC_24, Direct_24,
                    DE_46, log2FC_46, Direct_46,
                    DE_68, log2FC_68, Direct_68,
                    DE_28, log2FC_28, Direct_28)


test = at_least_one_sig_df %>% filter(
  DE_02 == "yes",
  DE_28 == "yes"
)
```

```{r}
list = read.table('../processed_data/degenes/down_02.txt')

```

```{r}
# no need to run this if already saved dataframe.
# lets look at these genes more closely, but first, get FPKM measurements for each gene.
ordered_geneLength = data.frame("Ensembl.Gene.ID" = rownames(assay(dds)), stringsAsFactors = FALSE) %>% 
    left_join(., gene_info, by = "Ensembl.Gene.ID")
mcols(dds)$basepairs = ordered_geneLength$gene_length 
fpkm_dds <- fpkm(dds)

ID_mapping = read.table('../raw_data/Mappings/rnaseq_mapping_list.txt', header = TRUE)

all_genes <- as.data.frame(fpkm_dds) %>% 
  tibble::rownames_to_column(var = "Ensembl.Gene.ID") %>%
  left_join(., gene_info, by = "Ensembl.Gene.ID") %>%
  dplyr::select(-c("Ensembl.Gene.ID", "gene_length")) %>%
  distinct(Associated.Gene.Name, .keep_all = TRUE) %>%
  tibble::column_to_rownames(var = "Associated.Gene.Name") %>%
  t() %>% as.data.frame() %>%
  tibble::rownames_to_column(var = "RNA_seq_ID") %>%
  left_join(., ID_mapping, by = "RNA_seq_ID") %>%
  relocate(Lipidomics_ID) %>%
  separate(Lipidomics_ID, c("Treatment", "ID"), sep = "_") %>%
  mutate(Treatment = ifelse(Treatment == 'CHOW8', 'Chow8', Treatment)) %>%
  dplyr::select(-RNA_seq_ID) %>%
  relocate(ID)

write.csv(all_genes, '../processed_data/datasets/RNAseq_FPKM.csv', row.names = F)
```

```{r}
# START HERE FOR DOWNSTREAM ANALYSES
all_genes = read.csv('../processed_data/datasets/RNAseq_FPKM.csv', header = TRUE, check.names = FALSE)
```

```{r}
all_genes %>% 
  dplyr::select(Treatment, Hmgcr, Cidea, Enho, Hal, Cd36, Cfd, Col1a1) %>%
  gather(Feature, Value, -Treatment) %>%
  ggplot(aes(y = log10(Value+1), x = Treatment)) + 
  geom_boxplot(outlier.shape = NA, aes(fill = Treatment), alpha = 0.6) + 
  geom_jitter(aes(color = Treatment), alpha = 0.7) +
  scale_color_manual(values = c('black', '#3b9bb3', '#75af4f', '#e39225', '#c13b41')) +
  scale_fill_manual(values = c('black', '#3b9bb3', '#75af4f', '#e39225', '#c13b41')) +
  facet_wrap(~Feature, scales = 'free') + 
  theme(axis.title.x = element_blank()) + 
  labs(y = 'log10(FPKM + 1)')
```