---
title: "Analysis of RNA-seq data"
author: "Timothy Yu"
output: 
  github_document:
    toc: true
    html_preview: false
editor_options:
  chunk_output_type: inline
---
This notebook analyzes the RNA-seq dataset.
```{r, include = FALSE}
library(dplyr)
library(tidyr)
library(ggplot2)
require(cowplot)
theme_set(theme_cowplot())
library(viridis)
library(RColorBrewer)
library(DESeq2)
library(GenomicFeatures)
library(apeglm)
library(EnhancedVolcano)
```

```{r}
sessionInfo()
```

```{r}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(dev.args = list(png = list(type = "cairo")))
```

Load in all sample files.
Note: one outlier `7B_Index_09_control8` was removed after visualizing PCA. If want to visualize it without that sample, do not run line of code where it removes file name from `sampleFiles`.
```{r}
directory <- "../raw_data/ETV007_RNAseq_counts"
sampleFiles <- (list.files(directory))
sampleFiles <- sampleFiles[-33]

# load in the gene key information
key = read.table('../raw_data/Mappings/ensembl84_info.txt', header = TRUE, sep = '\t', quote="", fill=FALSE, stringsAsFactors = FALSE)
```

Get gene lengths for downstream FPKM normalization.
```{r}
gene_info = key %>% mutate(gene_length = Gene.End..bp. - Gene.Start..bp. + 1) %>% dplyr::select(Ensembl.Gene.ID, Associated.Gene.Name, gene_length) %>% distinct()
```

Create `DESeq` object.
```{r}
sampleCondition <- sub("(.*treated).*","\\1",sampleFiles)
sampleTable <- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleFiles)

sampleTable <- sampleTable %>% 
  separate(col = 'condition', into = c("exp", "index", "condition"), sep = "_", remove = F)
sampleTable <- subset(sampleTable, select = -c(exp, index))
sampleTable$condition <- gsub(".count", "", sampleTable$condition)
sampleTable$condition <- gsub("control8", "0", sampleTable$condition)
sampleTable$condition <- gsub("treated", "", sampleTable$condition)

ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                       directory = directory,
                                       design = ~condition)

#Set the reference group to 0 week
ddsHTSeq$condition <- relevel(ddsHTSeq$condition, ref = "0")

#Filter out reads whose row sums are less than 10
ddsHTSeq <- ddsHTSeq[rowSums(counts(ddsHTSeq))>=10,]

# Run DESeq
dds <- DESeq(ddsHTSeq)
```

PCA of variance-stabilized transformed data.
```{r, rna1, echo=T, fig.width=6, fig.height=4, fig.align="center", dpi=300, dev="png"}
vsd = vst(dds)
levels(vsd$condition) = c("CHOW8", "WD2", "WD4", "WD6", "WD8")
plotPCA(vsd) + scale_color_manual(values = c('black', '#3b9bb3', '#75af4f', '#e39225', '#c13b41'), name = "Treatment")

#ggsave('../figures/rnaseq_pca.pdf', height = 4.5, width = 5)
```
Check normalized data.
```{r}
raw <- reshape2::melt(assay(dds))
colnames(raw) <- c("Gene","Sample","Expression")
ggplot(raw, aes(x=Sample, y=Expression)) + geom_violin() + theme(axis.text.x = element_text(angle=45, hjust=1))

matrix <- reshape2::melt(assay(vsd))
colnames(matrix) <- c("Gene","Sample","Expression")
ggplot(matrix, aes(x=Sample, y=Expression)) + geom_violin() + theme(axis.text.x = element_text(angle=45, hjust=1))
```

Generate all volcano plots. We use `ashr` to shrink fold-changes.
Vertical lines indicate |log2FC| > 0.5.
Horizontal lines indicate padj < 0.05.
```{r}
mapping = key %>% dplyr::select(Ensembl.Gene.ID, Associated.Gene.Name) %>% distinct()

for(i in c(0, 2, 4, 6)) {
  for (j in c(2, 4, 6, 8)) {
    if(j > i) {
      res = lfcShrink(dds, contrast = c("condition", j, i), type = 'ashr')
      annotation = data.frame("Ensembl.Gene.ID" = rownames(res), stringsAsFactors = FALSE) %>% 
      left_join(., mapping, by = "Ensembl.Gene.ID") 
      
      # create custom key-value pairs for 'up-regulated', 'down-regulated', 'non-significant'
      keyvals <- ifelse(
        abs(res$log2FoldChange) > 0.5 & res$padj < 0.05, '#FF5A3F', ifelse(
          res$padj < 0.05, '#2A72FF', 'gray40'
        )
      )
      keyvals[is.na(keyvals)] <- 'gray40'
      names(keyvals)[keyvals == '#2A72FF'] <- 'padj < 0.05'
      names(keyvals)[keyvals == '#FF5A3F'] <- 'padj < 0.05 and log2FC > 0.5'
      names(keyvals)[keyvals == 'gray40'] <- 'NS'
      
      range = max(abs(min(res$log2FoldChange)), abs(max(res$log2FoldChange))) + 1
      
      EnhancedVolcano(res, 
                lab = annotation$Associated.Gene.Name, 
                labSize = 6,
                x = 'log2FoldChange', 
                y = 'padj',
                xlim = c(-range, range),
                title = paste(i, 'vs.', j), 
                pCutoff = 0.05, 
                FCcutoff = 0.5, 
                colCustom = keyvals,
                colAlpha = 0.75, 
                pointSize = 3.0)
      ggsave(paste('../figures/volcano/', i, 'vs', j, '.pdf', sep=''), height = 7.5, width = 6.2)
    }
  }
}
```

Obtain log2FC, padj, and direction of FC for each gene between each time-point comparison.
Calculate the number of DE genes between each time-point comparison.
DE means padj < 0.05 and |log2FC| > 0.5.
```{r, warning=FALSE}
mapping = key %>% dplyr::select(Ensembl.Gene.ID, Associated.Gene.Name) %>% distinct()
class_df = data.frame(matrix(NA, ncol=1, nrow=nrow(annotation)))[-1]
summary_df = data.frame("t1"=integer(), "t2"=integer(), "Total" =integer(), 
                        "Up"=integer(), "Down"=integer(), stringsAsFactors=FALSE)

for(i in c(0, 2, 4, 6)) {
  for (j in c(2, 4, 6, 8)) {
    if(j > i) {
      res = lfcShrink(dds, contrast = c("condition", j, i), type = 'ashr')
      fcname = paste('log2FC_', i, j, sep='')
      dename = paste('DE_', i, j, sep='')
      dirname = paste('Direct_', i, j, sep='')
  
      genes = as.data.frame(res) %>% 
        tibble::rownames_to_column(var = "Ensembl.Gene.ID") %>% 
        left_join(., mapping, by = "Ensembl.Gene.ID") %>%
        mutate(
          !!fcname := log2FoldChange,
          !!dename := ifelse(padj < 0.05 & abs(log2FoldChange) > 0.5, "yes", "no"),
          !!dirname := ifelse(log2FoldChange > 0, "up", "down")
        ) %>%
        dplyr::select(Associated.Gene.Name, !!as.symbol(fcname), 
                      !!as.symbol(dename), !!as.symbol(dirname))
  
      if(ncol(class_df) == 0) {
        class_df = cbind(class_df, genes)
      } else {
        stopifnot(identical(class_df$Associated.Gene.Name, genes$Associated.Gene.Name))
        class_df = cbind(class_df, genes %>% dplyr::select(-Associated.Gene.Name))
      }
      
      num_total = nrow(filter(genes, !!as.symbol(dename) == "yes"))
      num_up = nrow(filter(genes, !!as.symbol(dename) == "yes", !!as.symbol(dirname) == "up"))
      num_down = nrow(filter(genes, !!as.symbol(dename) == "yes", !!as.symbol(dirname) == "down"))
      num_category = data.frame("t1"=i, "t2"=j, "Total"=num_total, "Up"=num_up, 
                              "Down"=num_down, stringsAsFactors=FALSE)
      summary_df = rbind(summary_df, num_category)
    }
  }
}
```
Plot heat maps of significantly DE genes (padj < 0.05, |log2FC| > 0.5).
```{r, rna2, echo=T, fig.width=9, fig.height=3, fig.align="center", dpi=300, dev="png"}
a <- summary_df %>% 
  ggplot(aes(x=t1, y=t2)) + 
  geom_tile(aes(fill=Total), color='black', size=0.5, alpha = 0.9) + 
  scale_fill_gradient(low="#f1f1f1", high="#696969") + 
  scale_x_continuous(breaks=seq(0,8,2)) + 
  scale_y_continuous(breaks=seq(0,8,2)) + 
  labs(x="Weeks on Western Diet", y="Weeks on Western Diet", title="Total") + 
  geom_text(aes(label=Total), size=6) + 
  theme(legend.position="none")

b <- summary_df %>% 
  ggplot(aes(x=t1, y=t2)) + 
  geom_tile(aes(fill=Up), color='black', size=0.5, alpha=0.9) +
  scale_fill_gradient(low="#e5e5ff", high="blue") + 
  scale_x_continuous(breaks=seq(0,8,2)) + 
  scale_y_continuous(breaks=seq(0,8,2)) + 
  labs(x="Weeks on Western Diet", y="Weeks on Western Diet", title="Up-regulated") +
  geom_text(aes(label=Up), size=6) + 
  theme(legend.position="none")

c <- summary_df %>% 
  ggplot(aes(x=t1, y=t2)) + 
  geom_tile(aes(fill=Down), color='black', size=0.5, alpha=0.9) +
  scale_fill_gradient(low="#ffe5e5", high="red") +
  scale_x_continuous(breaks=seq(0,8,2)) + 
  scale_y_continuous(breaks=seq(0,8,2)) + 
  labs(x="Weeks on Western Diet", y="Weeks on Western Diet", title="Down-regulated") +
  geom_text(aes(label=Down), size=6) + 
  theme(legend.position="none")

plot_grid(a,b,c, ncol = 3)
# ggsave('../figures/rnaseq_pairwise_de_genes_heatmap_fccutoff.pdf', height = 3.7, width = 10)
rm(a,b,c)
```

The RNA-seq profiles have a largely `biphasic` response. We can categorize genes accordingly and count the number in each category.
```{r}
class_df = class_df %>% 
  mutate(biphasic_class = ifelse(DE_02 == 'yes' & DE_28 == 'no', 'uniq_02',
                 ifelse(DE_02 == 'no' & DE_28 == 'yes', 'uniq_28',
                 ifelse(DE_02 == 'yes' & DE_28 == 'yes', 'both', 'none'))))

print(nrow(filter(class_df, biphasic_class == 'uniq_02')))
print(nrow(filter(class_df, biphasic_class == 'uniq_28')))
print(nrow(filter(class_df, biphasic_class == 'both'))) 
print(nrow(filter(class_df, biphasic_class == 'none'))) 
```
Save lists of DE genes that fall into each category for downstream KEGG pathway enrichment analysis.
```{r}
save_list <- function(l, fname) {
  write.table(
    l,
    paste0('../processed_data/degenes/', fname, '.txt'),
    quote = F, col.names = F, row.names = F
  )
}

# DE exclusively between 02, up
up_02_list = (
  class_df %>% 
    filter(biphasic_class == 'uniq_02', Direct_02 == 'up') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(up_02_list, 'up_02')

# DE exclusively between 02, down
down_02_list = (
  class_df %>% 
    filter(biphasic_class == 'uniq_02', Direct_02 == 'down') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(down_02_list, 'down_02')

# DE exclusively between 28, up
up_28_list = (
  class_df %>% 
    filter(biphasic_class == 'uniq_28', Direct_28 == 'up') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(up_28_list, 'up_28')

# DE exclusively between 28, down
down_28_list = (
  class_df %>% 
    filter(biphasic_class == 'uniq_28', Direct_28 == 'down') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(down_28_list, 'down_28')

# DE between 02 (up) and 28 (up) 
up02_up28_list = (
  class_df %>% 
    filter(biphasic_class == 'both', Direct_02 == 'up', Direct_28 == 'up') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(up02_up28_list, 'up02_up28')

# DE between 02 (up) and 28 (down) 
up02_down28_list = (
  class_df %>% 
    filter(biphasic_class == 'both', Direct_02 == 'up', Direct_28 == 'down') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(up02_down28_list, 'up02_down28')

# DE between 02 (down) and 28 (up) 
down02_up28_list = (
  class_df %>% 
    filter(biphasic_class == 'both', Direct_02 == 'down', Direct_28 == 'up') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(down02_up28_list, 'down02_up28')

# DE between 02 (down) and 28 (down) 
down02_down28_list = (
  class_df %>% 
    filter(biphasic_class == 'both', Direct_02 == 'down', Direct_28 == 'down') %>% 
    dplyr::select(Associated.Gene.Name)
)
save_list(down02_down28_list, 'down02_down28')
```

We input the above gene lists into KEGG pathway enrichment via `https://biit.cs.ut.ee/gprofiler/gost`. We set organism to Mouse.

We then filter out the outputted KEGG pathways by:
1. any obvious irrelevant pathways (i.e., cancer, viral infection, etc.)

Ran on 5/22/23. These are in files `kegg_02_up.csv`, `kegg_02_down.csv`, `kegg_28_up.csv`, and `kegg_28_down.csv`.

```{r}
kegg_02_up = read.csv('../processed_data/degenes/kegg_02_up.csv', header = TRUE) %>% mutate(
  significance = negative_log10_of_adjusted_p_value, 
  direction = "up"
)

kegg_02_down = read.csv('../processed_data/degenes/kegg_02_down.csv', header = TRUE) %>% mutate(
  significance = -negative_log10_of_adjusted_p_value, 
  direction = "down"
)

kegg_28_up = read.csv('../processed_data/degenes/kegg_28_up.csv', header = TRUE) %>% mutate(
  significance = negative_log10_of_adjusted_p_value, 
  direction = "up"
)

kegg_28_down = read.csv('../processed_data/degenes/kegg_28_down.csv', header = TRUE) %>% mutate(
  significance = -negative_log10_of_adjusted_p_value, 
  direction = "down"
)

kegg_02_df = rbind(kegg_02_up, kegg_02_down) %>%
  filter(!term_name %in% c('Chemical carcinogenesis - DNA adducts',
                         'Chemical carcinogenesis - receptor activation',
                         'Drug metabolism - other enzymes',
                         'Drug metabolism - cytochrome P450',
                         'Metabolism of xenobiotics by cytochrome P450',
                         'Measles',
                         'Hepatitis C',
                         'Acute myeloid leukemia',
                         'Staphylococcus aureus infection',
                         'Epstein-Barr virus infection',
                         'Coronavirus disease - COVID-19'))

kegg_28_df = rbind(kegg_28_up, kegg_28_down) %>%
  filter(!term_name %in% c('Human papillomavirus infection',  'Amoebiasis'))

```

```{r, rna3, echo=T, fig.width=6, fig.height=8, fig.align="center", dpi=300, dev="png"}
kegg_02_df %>%
  ggplot(aes(y = significance, x = reorder(term_name, -significance))) +
  geom_bar(aes(fill = direction), stat='identity', color='black', width = 0.75) +
  scale_fill_manual(values=c('blue', 'red')) +
  coord_flip() +
  ylim(-15,15) +
  theme(legend.position = 'none') +
  labs(x="", y='-log10(padj)', title='0 vs. 2 weeks WD')
#ggsave('../figures/rnaseq_02_kegg.pdf', height = 6, width = 7)
```

```{r, rna4, echo=T, fig.width=6, fig.height=8, fig.align="center", dpi=300, dev="png"}
kegg_28_df %>%
  ggplot(aes(y = significance, x = reorder(term_name, -significance))) +
  geom_bar(aes(fill = direction), stat='identity', color='black', width = 0.75) +
  scale_fill_manual(values=c('blue', 'red')) +
  coord_flip() +
  ylim(-15,15) +
  theme(legend.position = 'none') +
  labs(x="", y='-log10(padj)', title='2 vs. 8 weeks WD')
#ggsave('../figures/rnaseq_28_kegg.pdf', height = 6, width = 8)
```

This chunk calculates and writes FPKM measurements for all genes.
```{r}
# no need to run this if already saved dataframe.
ordered_geneLength = data.frame("Ensembl.Gene.ID" = rownames(assay(dds)), stringsAsFactors = FALSE) %>% 
    left_join(., gene_info, by = "Ensembl.Gene.ID")
mcols(dds)$basepairs = ordered_geneLength$gene_length 
fpkm_dds <- fpkm(dds)

ID_mapping = read.table('../raw_data/Mappings/rnaseq_mapping_list.txt', header = TRUE)

all_genes <- as.data.frame(fpkm_dds) %>% 
  tibble::rownames_to_column(var = "Ensembl.Gene.ID") %>%
  left_join(., gene_info, by = "Ensembl.Gene.ID") %>%
  dplyr::select(-c("Ensembl.Gene.ID", "gene_length")) %>%
  distinct(Associated.Gene.Name, .keep_all = TRUE) %>%
  tibble::column_to_rownames(var = "Associated.Gene.Name") %>%
  t() %>% as.data.frame() %>%
  tibble::rownames_to_column(var = "RNA_seq_ID") %>%
  left_join(., ID_mapping, by = "RNA_seq_ID") %>%
  relocate(Lipidomics_ID) %>%
  separate(Lipidomics_ID, c("Treatment", "ID"), sep = "_") %>%
  mutate(Treatment = ifelse(Treatment == 'CHOW8', 'Chow8', Treatment)) %>%
  dplyr::select(-RNA_seq_ID) %>%
  relocate(ID)

# write.csv(all_genes, '../processed_data/datasets/RNAseq_FPKM.csv', row.names = F)
```

```{r}
# rmarkdown::render("rnaseq.Rmd")
# mv rnaseq.md ../markdowns/
# mv rnaseq_files ../markdowns/
```